src/basic/string-util.c

```diff
@@ -1187,6 +1187,41 @@ char *string_replace_char(char *str, char old_char, char new_char) {
         return str;
 }
 
+int make_cstring(const char *s, size_t n, MakeCStringMode mode, char **ret) {
+        char *b;
+
+        assert(s || n == 0);
+
```

Comment:
assert(ret)

Reply:
still missing?

Reply:
i instead add guards further down so that we handle ret == 0 properly, so that you can use the function for validation still.